### ISSUE TITLE
Model validation

### ISSUE THREAD

**Issue by clgiebink** [clgiebink/UT_FVS#11]
This thread will start a discussion on how model validation for the growth models will proceed. Some things to consider:

- What will be the training set?
- What will be the testing set?
- If the testing set is new trees, we will use population level estimates (no random effects).

**Comment by clgiebink** [comment]
What was discussed with Justin and John in Logan, UT:

The training set is my current data set, including the newly verified cores. With the new data, I'll have 100 or more cores per species. My testing set is the DBH remeasurements from the FIADB.

I can run several validation sets. What we've considered so far is growing the trees forward with various models. Results will be compared to the observations.

1. Compare projections to observations with and without tree-ring data (no climate parameters). Projections with tree-rings will use the models parameterized with my current data set. Projections without tree-rings will use the current fvs models. Right?
2. Compare projections to observations with and without climate (tree-ring data included). 

Still, some questions remain:

- In both scenarios, will we use population level estimates? It probably depends on if any of the trees in the data set were remeasured.

**Comment by clgiebink** [comment]
I've been thinking about how I will grow trees forward using the new models. In order to calculate and project with density metrics (CCF, BAL, CR), I'll need to pull down all trees on the same plots as the subject tree. The first projection cycle will be easy. I'll pull down DBH for all trees from the FIADB. After the first year, I'll need some way of growing the non-subject trees, which could be other species (not ES, PP, or DF) or small trees. In those cases, do I:
1. Use current FVS growth models: large diameter growth model for large trees of different species and small diameter growth model for trees < 1''?
2. Interpolate growth (DBH) from remeasurement data. I believe there are nonlinear methods of doing this: [Cao_2000.pdf](https://github.com/clgiebink/UT_FVS/files/4187015/Cao_2000.pdf)
3. Grow trees by some arbitrary constant, which is kind of the opposite of what we did when back calculating non-subject trees.
4. Is there another method that would work?

**Comment by rjderose** [comment]
Hi Courtney,
For this question, I think it would probably be most appropriate to use option 1. It seems weird because we are trying to improve that model, but we aren't doing all the species, so for the ones we aren't testing, we could just say we're using the 'most current' method, until they get improved as well, or something like that. 
I'm definitely open to other options...

**Comment by clgiebink** [comment]
Thanks for your feedback, @rjderose. Margaret suggested the same - to go with option 1. If I use current FVS models, there's still the problem of them not being annual. To get stand density metric (CCF, BAL, CR), I need to know the DBH of each tree on the plot for the given year, so I need to know how much the non-focal trees grew in one year. How do I go from the 10-year FVS model to an annual model? It can't just be 1/10 of the current model coefficients, right?

I think [Cao_2000.pdf](https://github.com/clgiebink/UT_FVS/files/4240239/Cao_2000.pdf) gives a method of estimating annual parameters from periodic remeasurements and [Cao_Strub_2008.pdf](https://github.com/clgiebink/UT_FVS/files/4240246/Cao_Strub_2008.pdf) compares 4 methods for estimating annual stand density. Mark Castle shared these papers with me. This is the method he used for annualizing the Alaskan large tree growth model. So I guess it would be FVS's current method for annual growth models.

His adviser, Aaron Weiskittel, has also used it to make annual models: [Weiskittel_2007.pdf](https://github.com/clgiebink/UT_FVS/files/4240295/Weiskittel_2007.pdf) and [Weiskittel_2016.pdf](https://github.com/clgiebink/UT_FVS/files/4240298/Weiskittel_2016.pdf)

**Comment by clgiebink** [comment]
We're going to first try the simplest method: interpolating DBH for nonfocal trees. Trees are grow each year by a constant calculated by subtracting beginning of the projection cycle DBH minus end of the projection cycle DBH, divided by census interval or
`(fDIA - DIA)/census_interval`

A couple issues that have come up:
1. Some trees that are alive at the beginning of the projection cycle die, are cut, or are excluded from plot by the end of the projection cycle. For now, I'm excluding trees that are excluded from the remeasurement data and I am growing dead and cut trees until their estimated mortality year (MORTYR). But what if most of the plot dies during the projection cycle? I already filtered for undisturbed plots, but should I also filter for plots where basal area doesn't significantly decrease? It could perhaps cause a release in my focal tree.
2. Some trees shrink during the projection cycle. Margaret mentioned there might be a biological reason? What is this reason? Should I allow trees to shrink? Or add a species specific constant?
3. Some trees are on different conditions (CONDID = 1,2,3). How does this affect the projection?

**Comment by clgiebink** [comment]
Resolutions to the questions above:

1. Don't filter plots where basal area significantly decreases.
2. Allow trees to shrink.
3. Filter plots with more than one condition (CONDID > 1).
